src/app/core/params.ts

```diff
@@ -3,27 +3,22 @@ import Parameter from "./parameter";
  * Params class
  */
 export default class Params {
+    private unique: boolean;
+    private heritable: boolean;
+    parameters: any;
+    
+
     /**
      * Default constructor create a Params object.
      * @param {} values Values of the params
      * @param {Boolean} unique Boolean if it's unique
      * @param {} heritable Boolean if it's heritable
      * @param {} rawparameters ?
      */
-    constructor(values, unique, heritable, rawparameters) {
+    constructor(values: any, unique: boolean, heritable: boolean) { 
```

Comment:
`values` - dictionary of keys and values
`unique` - List of keys that are unique (list of strings)
`heritable` - List of keys which are unique (list of strings)